From 984cc45b711395004653f7e2bb6452f98f4d7471 Mon Sep 17 00:00:00 2001
From: Stefan Wiehler <stefan.wiehler@missinglinkelectronics.com>
Date: Tue, 29 May 2018 11:21:40 +0200
Subject: [PATCH 2/5] Enable link-time optimization

Signed-off-by: Stefan Wiehler <stefan.wiehler@missinglinkelectronics.com>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 04381b7a..0dc50244 100644
--- a/Makefile
+++ b/Makefile
@@ -155,10 +155,10 @@ CPPFLAGS		=	${DEFINES} ${INCLUDES} -nostdinc		\
 				-Wmissing-include-dirs -Werror
 ASFLAGS			+=	$(CPPFLAGS) $(ASFLAGS_$(ARCH))			\
 				-D__ASSEMBLY__ -ffreestanding 			\
-				-Wa,--fatal-warnings
+				-Wa,--fatal-warnings -flto
 TF_CFLAGS		+=	$(CPPFLAGS) $(TF_CFLAGS_$(ARCH))		\
 				-ffreestanding -fno-builtin -Wall -std=gnu99	\
-				-Os -ffunction-sections -fdata-sections
+				-Os -ffunction-sections -fdata-sections -flto
 
 GCC_V_OUTPUT		:=	$(shell $(CC) -v 2>&1)
 PIE_FOUND		:=	$(findstring --enable-default-pie,${GCC_V_OUTPUT})
@@ -167,7 +167,7 @@ ifeq ($(PIE_FOUND),1)
 TF_CFLAGS		+=	-fno-PIE
 endif
 
-TF_LDFLAGS		+=	-Wl,--fatal-warnings -O1 -nostdlib
+TF_LDFLAGS		+=	-Wl,--fatal-warnings -O1 -nostdlib -flto
 TF_LDFLAGS		+=	-Wl,--gc-sections -Wl,--build-id=none
 TF_LDFLAGS		+=	$(TF_LDFLAGS_$(ARCH))
 
-- 
2.23.0

